### ISSUE TITLE
[azure_rm] Specify a keyvault and secret.

### ISSUE THREAD

**Issue by volkdir** [hashicorp/packer#7460]
Feature request:
Due to security reasons in our Azure Subscription we are not allowed to create key vaults. That is only allowed for user with special rights. But we like to build prepared windows images in a automatic fashion and those user can't do it.
If we run our packer template we get this error:
```
atus=400 Code="InvalidTemplateDeployment" Message="Deployment failed with multi ple errors: 'Authorization failed for template resource '#########' of typ e 'Microsoft.KeyVault/vaults'. The client '##############'  with object id '##########' does not have permission  to perform action 'Microsoft.KeyVault/vaults/write' at scope '/subscriptions/###############/resourceGroups/packer-Resource-Group-ay5poget
--
```

I assume the vault will be used to specify a secret to connect to the VM. Maybe it will be possible to specify a prepared vault/key for that.

**Comment by nfischer2** [comment]
I am also experiencing a similar issue in our Azure subscriptions:  (Deny)","id":"/providers/Microsoft.Management/managementGroups/All/providers/Microsoft.Authorization/policyAssignments/Premium KeyVault"},"policyDefinition":{"name":"Require Premium KeyVault 

Our policy requires a Premium SKU KeyVault so attempts to create a Standard SKU KeyVault are denied.  There may be issues going forward with permissions to allow creation of a KeyVault.   It would be preferred to specify a KeyVault that Packer can use similar to specifying existing resource groups and vnets.  Due to our policies this affects all Windows builds and we will be unable to use Packer without the ability to use Premium SKU KeyVaults.

**Comment by SwampDragons** [comment]
I believe this is fixed by #8704; builds of the branch can be found here: https://circleci.com/gh/hashicorp/packer/32687#artifacts/containers/0

If I can answer any usage questions let me know (docs are in the PR). It works for me, but I am not completely certain that my toy keyvault implementation is the same as something we'd see in a real production usage so I'd like a sanity check from someone affected by the issue.
